packages/xrpl/test/integration/transactions/depositPreauth.test.ts

```diff
@@ -32,4 +36,54 @@ describe('DepositPreauth', function () {
     },
     TIMEOUT,
   )
+
+  it(
+    'credentials',
+    async () => {
+      const subjectWallet = await generateFundedWallet(testContext.client)
+
+      const credentialCreateTx: CredentialCreate = {
+        TransactionType: 'CredentialCreate',
+        Account: testContext.wallet.classicAddress,
+        Subject: subjectWallet.classicAddress,
+        CredentialType: stringToHex('Test Credential Type'),
+      }
+
+      await testTransaction(
+        testContext.client,
+        credentialCreateTx,
+        testContext.wallet,
+      )
+
+      const credentialAcceptTx: CredentialAccept = {
+        TransactionType: 'CredentialAccept',
+        Account: subjectWallet.classicAddress,
+        Issuer: testContext.wallet.classicAddress,
+        CredentialType: stringToHex('Test Credential Type'),
+      }
+
+      await testTransaction(
+        testContext.client,
+        credentialAcceptTx,
+        subjectWallet,
+      )
+
+      const authorizeCredentialObj: AuthorizeCredential = {
+        Credential: {
+          Issuer: testContext.wallet.classicAddress,
+          CredentialType: stringToHex('Test Credential Type'),
+        },
+      }
+
+      const wallet2 = Wallet.generate()
+      await fundAccount(testContext.client, wallet2)
+      const tx: DepositPreauth = {
+        TransactionType: 'DepositPreauth',
+        Account: testContext.wallet.classicAddress,
+        AuthorizeCredentials: [authorizeCredentialObj],
```

Comment:
This is nice work. However, for the purpose of testing the functionality of `DepositPreauth` transaction, there is no need to execute `CredentialCreate` or `CredentialAccept` transaction.

`DepositPreauth` cannot do sanity checks on the input Credentials.

Reply:
@khancode  that's a fair point 👍